src/pages/settings/Subscription/CardSection/utils.ts

```diff
@@ -0,0 +1,21 @@
+import {addMonths, format, isBefore} from 'date-fns';
+import CONST from '@src/CONST';
+
+/**
+ * Get the next billing date.
+ *
+ * @param initialDate - The initial billing date in 'yyyy-MM-dd' format.
+ * @returns - The next billing date in 'yyyy-MM-dd' format.
+ */
+function getNextBillingDate(initialDate: string): string {
```

Comment:
what do you think of this to get rid of `while` loop.

```
function getNextBillingDate(initialDate: string): string {
    const start = new Date(initialDate);
    const today = new Date();

    // Calculate the number of months difference and add one if start date is before today
    const monthsDiff = differenceInMonths(today, start);
    const nextBillingDate = addMonths(start, monthsDiff + 1);

    return format(nextBillingDate, CONST.DATE.MONTH_DAY_YEAR_FORMAT);
}

```

Reply:
Yeah if we can eliminate a loop that would be great!